src/status.js

```diff
@@ -56,11 +78,13 @@ export default class Status extends Component {
       content = tResults(length, contentSelectedOption)
     }
 
+    this.debounceContentUpdate()
```

Comment:
Is there a way to debounce the content update without the `render` function effectively (through a timeout) changing the component state and thus triggering another `render`?

We're wondering if there's a way to do it using the `componentWillUpdate` lifecycle event on the `Status` component?

Alternatively, avoid using the virtual DOM entirely, and update the status using the native DOM API?

Reply:
I actually had a few variations around the debouncing - including using the `componentWillUpdate` lifecycle method at one point. I settled on the current form as the tidiest / most readable - but no issue with moving to the lifecycle if preferred.
Whichever approach, I need to avoid creating an infinite loop by ensuring the state is only amended one more time.

There's also the option to bring in [this](https://github.com/samuherek/use-debounce-preact) library, in preference to the local debounce function. I haven't investigated so far, as the local solution performs ok, but it may improve readability.

The debounce is bringing a couple of benefits in terms of JAWS and rapid typing, and in preventing the typing echo interruption in VoiceOver - so moving to direct DOM manipulation wouldn't itself resolve those issues - although it might avoid the need for the duplicate live regions.

Reply:
I've experimented with the direct DOM update approach. Be good to discuss over hangout.
My changes for this one are parked on another branch - `markh-direct-dom`
It's the only change relating to these comments that I haven't applied directly to my master branch - pending discussion :)